SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -381,18 +381,7 @@ int64 USpatialActorChannel::ReplicateActor()
 	{
 		FOutBunch DummyOutBunch;
 
-		for (UActorComponent* ActorComponent : Actor->GetReplicatedComponents())
-		{
-			const FUnrealObjectRef ObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromObject(ActorComponent);
-
-			if (ObjectRef.IsValid())
-			{
-				FClassInfo& SubobjectInfo = Info.SubobjectInfo[ObjectRef.Offset].Get();
-
-				bWroteSomethingImportant |= ReplicateSubobject(ActorComponent, SubobjectInfo, RepFlags);
-				bWroteSomethingImportant |= ActorComponent->ReplicateSubobjects(this, &DummyOutBunch, &RepFlags);
-			}
-		}
+		bWroteSomethingImportant |= Actor->ReplicateSubobjects(this, &DummyOutBunch, &RepFlags);
```

Comment:
Just so I understand, AActor::ReplicateSubobjects calls UActorComponent::ReplicateSubobject which calls USpatialActorChannel::ReplicateSubobject (which takes in a bunch) and then calls USpatialActorChannel::ReplicateSubobject which does the actual Spatial replication?

What's the benefit of going this flow rather than explicitly having the code from before? I feel like this makes it harder to understand.

Reply:
Yes, actors and actor components need to manually call ReplicateSubobject on non component subobjects. This is needed so users can override Actor::ReplicateSubobjects and ensure their logic is executed in our stack.

Reply:
HmmmmmmmmMMMMMMMMMMMMMMM ok. Can you put my comment above as a comment in the code as the flow of what happens normally?